packages/contracts/src/utils/log_decoder.ts

```diff
@@ -0,0 +1,29 @@
+import { LogWithDecodedArgs, RawLog } from '@0xproject/types';
+import { AbiDecoder } from '@0xproject/utils';
+import * as _ from 'lodash';
+import * as Web3 from 'web3';
+
+import { artifacts } from './artifacts';
+import { Artifact } from './types';
+
+export class LogDecoder {
+    private _abiDecoder: AbiDecoder;
+    constructor(networkId: number) {
```

Comment:
This should be `networkIdIfExists?: number`, if the value can every be `undefined`.

Reply:
I think since this is an immediate validation and throw that `networkIdIfExists?` isn't necessary.  Perhaps if undefined here was valid option and could be handled, not thrown.

Reply:
There is no guarentee this validation will stay in place. I want the semantic meaning of this arg to be: "hey, watch out, this might be passed in as undefined".